### PR TITLE
Bug 1443667 - Fix 'tier' not yet assigned when getJobGroupInfo() creates map

### PR DESCRIPTION
**Pull request by camd**
I went ahead and optimized the deconstruction, too.  Since this function is called a lot, every bit helps for speed.  :)

### CODE REVIEW

File: ui/js/models/resultsets_store.js [hunk 1]
```diff
@@ -910,20 +910,18 @@ treeherder.factory('ThResultSetStore', [
...
-            }
-            return { name: name, tier: tier, symbol: symbol, mapKey: mapKey };
+        const getJobGroupInfo = function (job) {
+          const { job_group_name, job_group_symbol, platform, platform_option, result_set_id, tier } = job;
+          let updatedTier;
+          let updatedSymbol = job_group_symbol;
+
+          if (tier && tier !== 1) {
+            updatedSymbol = job_group_symbol === '?' ? '' : job_group_symbol;
+            updatedTier = tier;
```
Comment (edmorley) [inline ui/js/models/resultsets_store.js]: `updatedTier` is always `null` unless tier was 2/3? I'm not sure I follow the reason why this function did it like this?

Comment (camd) [reply ui/js/models/resultsets_store.js]: You make a really good point.  Yeah, this is some funky logic.  And the decision on whether or not to display a ``tier`` should be in the front-end code, not here.  So moved it out.

**Comment by camd** [comment]
I also found some dead code, so took the liberty to rip it out while I was here.  I can move to a new PR if you prefer.

**Comment by camd** [comment]
hold off on this for a few more minutes.  I'm still working...  :)

**Pull request merged**